src/main/java/ca/corefacility/bioinformatics/irida/service/analysis/workspace/galaxy/AnalysisWorkspaceServiceGalaxy.java

```diff
@@ -138,8 +141,16 @@ private AnalysisOutputFile buildOutputFile(String analysisId, String labelPrefix
 		String datasetId = dataset.getId();
 		String fileName = dataset.getName();
 
+		Path tmpOutputFile = Files.createTempFile(dataset.getName(), null);
+		tmpOutputFile.toFile().deleteOnExit();
+
+		galaxyHistoriesService.downloadDatasetTo(analysisId, datasetId, tmpOutputFile);
 		Path outputFile = outputDirectory.resolve(fileName);
-		galaxyHistoriesService.downloadDatasetTo(analysisId, datasetId, outputFile);
+		if (FileUtilities.isZippedFile(tmpOutputFile)) {
```

Comment:
So I'm wondering if we actually need to rename output files to add `.zip` here? Could we not look for an `.html` extension when loading the file in the **Analysis** page and open it up to check if it's a zip file (without needing it to end in `.html.zip`)? I guess it messes up with downloading the data (since you'll download a `.html` file even though it's a `.zip` file, but maybe we could modify the code for downloading output files to append a `.zip` there?

The reason I am thinking it would be nice to avoid re-naming files when saving, is that then this code to extract HTML files from a zip `.html` archive will work with all existing analysis submissions. In particular, the **Assembly Annotation Collection** pipeline already has QUAST results in a `.html` zip archive and if we remove any re-naming of output files from this code we could also just automatically display QUAST results from previous runs.

Reply:
I addressed that in 4c57b66 and in 951c063 I changed `getLabel` in `AnalysisOutputFile` to append zipped html files with `.zip` so that downloading works as expected.

Reply:
Awesome. Thanks.